packages/medusa/src/api/routes/admin/inventory-items/index.ts

```diff
@@ -0,0 +1,140 @@
+import { Router } from "express"
+import "reflect-metadata"
+import { DeleteResponse, PaginatedResponse } from "../../../../types/common"
+import { InventoryItemDTO } from "../../../../types/inventory"
+import middlewares, {
+  transformBody,
+  transformQuery,
+} from "../../../middlewares"
+import { AdminGetInventoryItemsParams } from "./list-inventory-items"
+import { AdminGetInventoryItemsItemParams } from "./get-inventory-item"
+import { AdminPostInventoryItemsInventoryItemReq } from "./update-inventory-item"
+import { AdminGetInventoryItemsItemLocationLevelsParams } from "./list-location-levels"
+import {
+  AdminPostInventoryItemsItemLocationLevelsReq,
+  AdminPostInventoryItemsItemLocationLevelsParams,
+} from "./create-location-level"
+import {
+  AdminPostInventoryItemsItemLocationLevelsLevelReq,
+  AdminPostInventoryItemsItemLocationLevelsLevelParams,
+} from "./update-location-level"
+import { checkRegisteredModules } from "../../../middlewares/check-registered-modules"
+
+const route = Router()
+
+export default (app) => {
+  app.use(
+    "/inventory-items",
+    checkRegisteredModules({
+      inventoryService:
+        "Inventory is not enabled. Please add an Inventory module to enable this functionality.",
+    }),
+    route
+  )
+
+  route.get(
+    "/",
+    transformQuery(AdminGetInventoryItemsParams, {
+      defaultFields: defaultAdminInventoryItemFields,
+      defaultRelations: defaultAdminInventoryItemRelations,
+      isList: true,
+    }),
+    middlewares.wrap(require("./list-inventory-items").default)
+  )
+
+  route.post(
+    "/:id",
+    transformQuery(AdminGetInventoryItemsItemParams, {
+      defaultFields: defaultAdminInventoryItemFields,
+      defaultRelations: defaultAdminInventoryItemRelations,
+      isList: false,
+    }),
+    transformBody(AdminPostInventoryItemsInventoryItemReq),
+    middlewares.wrap(require("./update-inventory-item").default)
+  )
+
+  route.post(
+    "/:id/location-levels",
+    transformQuery(AdminPostInventoryItemsItemLocationLevelsParams, {
+      defaultFields: defaultAdminInventoryItemFields,
+      defaultRelations: defaultAdminInventoryItemRelations,
+      isList: false,
+    }),
+    transformBody(AdminPostInventoryItemsItemLocationLevelsReq),
+    middlewares.wrap(require("./create-location-level").default)
+  )
+
+  route.get(
+    "/:id/location-levels",
+    transformQuery(AdminGetInventoryItemsItemLocationLevelsParams, {
+      defaultFields: defaultAdminInventoryItemFields,
+      defaultRelations: defaultAdminInventoryItemRelations,
+      isList: false,
+    }),
+    middlewares.wrap(require("./list-location-levels").default)
+  )
+
+  route.delete(
+    "/:id/location-levels/:location_id",
+    middlewares.wrap(require("./delete-location-level").default)
```

Comment:
todo: missing transformQuery

Reply:
wont add query params to the delete method... removed all references of that

Reply:
Sorry miss read that it was the delete one 👌